### PR TITLE
Add regx pattern file filter for distributed load

### PR DESCRIPTION
**Pull request by JiamingMai**
Add regx pattern file filter for distributed load.

**Example:**
The following request allows us to load the files under `/test-load` directory with "hello" prefix:
`curl -X GET http://localhost:28080/v1/load?path=s3a://jiamingmai-test/test-load&opType=submit&verbose=true&fileFilterRegx=^hello.*`

### CODE REVIEW

File: dora/core/server/master/src/main/java/alluxio/master/job/LoadJobFactory.java [hunk 1]
```diff
@@ -60,11 +62,23 @@ public Job<?> create() {
         .ofNullable(AuthenticatedClientUser.getOrNull())
         .map(User::getName);
 
+    Predicate<UfsStatus> predicate = Predicates.alwaysTrue();
```
Comment (jja725) [inline dora/core/server/master/src/main/java/alluxio/master/job/LoadJobFactory.java:65]: These logic should also appear at JournaledLoadJobFactory so we can recover from journal entry

File: dora/core/server/master/src/main/java/alluxio/master/job/LoadJobFactory.java [hunk 2]
```diff
@@ -60,11 +62,23 @@ public Job<?> create() {
...
+      }
+    }
+
     UnderFileSystem ufs = mFs.getUfsManager().getOrAdd(new AlluxioURI(path),
         UnderFileSystemConfiguration.defaults(Configuration.global()));
     Iterable<UfsStatus> iterable = new UfsStatusIterable(ufs, path,
         Optional.ofNullable(AuthenticatedClientUser.getOrNull()).map(User::getName),
-        Predicates.alwaysTrue());
+        predicate);
     return new DoraLoadJob(path, user, UUID.randomUUID().toString(), bandwidth, partialListing,
```
Comment (jja725) [inline dora/core/server/master/src/main/java/alluxio/master/job/LoadJobFactory.java:82]: We also want to put this filter information into DoraLoadJob so we can turn this job into a journal entry.

**Review by jja725 — CHANGES_REQUESTED** [review]
Mostly LGTM. We need to add these filter information into journal entry so we can recover from crash. Also can we add a test for this? We can try to follow loadcommandTest

File: dora/core/server/master/src/main/java/alluxio/master/job/LoadJobFactory.java (hunk 1 above)
Comment (JiamingMai) [reply dora/core/server/master/src/main/java/alluxio/master/job/LoadJobFactory.java:65]: Updated. I put these logic into JournaledLoadJobFactory.

File: dora/core/server/master/src/main/java/alluxio/master/job/LoadJobFactory.java (hunk 2 above)
Comment (JiamingMai) [reply dora/core/server/master/src/main/java/alluxio/master/job/LoadJobFactory.java:84]: Updated.

**Comment by JiamingMai** [comment]
> Mostly LGTM. We need to add these filter information into journal entry so we can recover from crash. Also can we add a test for this? We can try to follow loadcommandTest

Updated. Tests are added in `LoadCommandTest`.

**Review by jja725 — COMMENTED** [review]
LGTM, one minor test verification needed

File: dora/tests/integration/src/test/java/alluxio/client/cli/fs/command/DoraLoadCommandIntegrationTest.java [hunk 3]
```diff
@@ -83,4 +83,32 @@ public void testCommand() throws Exception {
...
+
+    assertEquals(0, mFsShell.run("load", path, "--submit", "--verify",
+        "--file-filter-regx", ".*B"));
+    assertEquals(0, mFsShell.run("load", path, "--progress"));
+    while (!mOutput.toString().contains("SUCCEEDED")) {
+      assertEquals(0, mFsShell.run("load", path, "--progress"));
+      Thread.sleep(1000);
+    }
+    assertTrue(mOutput.toString().contains("Inodes Processed: 1"));
+  }
```
Comment (jja725) [inline dora/tests/integration/src/test/java/alluxio/client/cli/fs/command/DoraLoadCommandIntegrationTest.java:113]: verify only B is loaded, not A&C

**Review by jja725 — APPROVED** [review]
LGTM

File: dora/tests/integration/src/test/java/alluxio/client/cli/fs/command/DoraLoadCommandIntegrationTest.java (hunk 3 above)
Comment (JiamingMai) [reply dora/tests/integration/src/test/java/alluxio/client/cli/fs/command/DoraLoadCommandIntegrationTest.java:116]: Updated.

**Comment by JiamingMai** [comment]
alluxio-bot, merge this please

**Pull request merged**